pkg/operator/ceph/cluster/mon/config.go

```diff
@@ -373,18 +334,24 @@ func ExtractKey(contents string) (string, error) {
 	return secret, nil
 }
 
-// PopulateExternalClusterInfo Add validation in the code to fail if the external cluster has no OSDs keep waiting
-func PopulateExternalClusterInfo(context *clusterd.Context, ctx context.Context, namespace string, ownerInfo *k8sutil.OwnerInfo) *cephclient.ClusterInfo {
+// PopulateExternalClusterInfo Add validation in the code to fail if the external cluster has no
+// OSDs keep waiting
+func PopulateExternalClusterInfo(context *clusterd.Context, ctx context.Context, namespace string, ownerInfo *k8sutil.OwnerInfo) (*cephclient.ClusterInfo, error) {
 	for {
-		clusterInfo, _, _, err := LoadClusterInfo(context, ctx, namespace)
+		// Checking for the context makes sure we don't loop forever with a canceled context
+		if ctx.Err() != nil {
+			return nil, ctx.Err()
+		}
+		clusterInfo, _, _, err := CreateOrLoadClusterInfo(context, ctx, namespace, nil)
```

Comment:
Should this not remain as LoadClusterInfo()? We don't ever want to create cluster info for an external cluster, right?
```suggestion
		clusterInfo, _, _, err := LoadClusterInfo(context, ctx, namespace, nil)
```

Reply:
Actually `LoadClusterInfo()` just calls `CreateOrLoadClusterInfo()` with a nil owerinfo. So in the end it's the same.